File: src/manual.rs
use crate::{utils::{extract_mc_version, get_versions}, Args};



pub fn main(args: Args) {

    let version: String;
    if let Some(v) = args.version_ {
        version = v;
    } else {
        eprintln!("No version specified. Please provide a version.");
        std::process::exit(1);
    }

    let versions = get_versions("@minecraft/server");

    

    // println!("Versions: {:?}", versions);

    if let Some(version) = versions.iter().find(|&v| *v == version) {
        println!("Found version: {}", version);
    } else if let Some(version) = versions.iter().find(|&v| v.contains(version.as_str())) {
        let version = extract_mc_version(version);
        println!("Found version: {}", version);

        let versions = get_versions("@minecraft/server");
        let server = versions.iter().find(|&v| v.contains(&version)).unwrap();
        let versions = get_versions("@minecraft/server-ui");
        // println!("Versions: {:?}", versions);
        let server_ui = versions.iter().find(|&v| v.contains(&version)).unwrap();

        println!("Server version: {}", server);
        println!("Server UI version: {}", server_ui);

    } else {
        eprintln!("Version not found: {}", version);
        std::process::exit(1);
    }
}

File: src/commands/manual.ts
import { getVersions, extractMcVersion } from "../utils/versions.js";
import * as logger from "../utils/logger.js";

export async function manual(version: string): Promise<void> {
    const versions = await getVersions("@minecraft/server");

    // Exact match
    const exact = versions.find((v) => v === version);
    if (exact) {
        logger.info(`Found exact version: ${exact}`);
        return;
    }

    // Partial match
    const partial = versions.find((v) => v.includes(version));
    if (!partial) {
        logger.error(`Version not found: ${version}`);
        process.exit(1);
    }

    const mcVersion = extractMcVersion(partial);
    logger.info(`Found version: ${mcVersion}`);

    const serverVersions = await getVersions("@minecraft/server");
    const server = serverVersions.find((v) => v.includes(mcVersion));

    const serverUiVersions = await getVersions("@minecraft/server-ui");
    const serverUi = serverUiVersions.find((v) => v.includes(mcVersion));

    if (server) logger.info(`Server version: ${server}`);
    if (serverUi) logger.info(`Server UI version: ${serverUi}`);
}
